### PR TITLE
Created team members list and card

### PR DESCRIPTION
**Pull request by CiprianDumitriu**
Created team members list and card together with the tests needed for them.

### CODE REVIEW

File: cypress/integration/team/list.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  before(() => {
+      cy.task('db:seedTeamMembers')
+    }
+  )
+  it('Should not have permissions when not logged in', () => {
+    cy.visit('http://localhost:3000/team-members/list')
+
+    cy.contains('admin user').should('not.exist')
+    cy.contains('Welcome to SalesWell')
+  })
```
Comment (keepcoding-tech) [inline cypress/integration/team/list.js]: Let's put an empty line between the functions

File: cypress/integration/team/list.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+describe('List team members', () => {
+  before(() => {
+      cy.task('db:seedTeamMembers')
+    }
+  )
+  it('Should not have permissions when not logged in', () => {
+    cy.visit('http://localhost:3000/team-members/list')
+
+    cy.contains('admin user').should('not.exist')
```
Comment (keepcoding-tech) [inline cypress/integration/team/list.js]: This test is not completely correct, can you find out why?

File: cypress/integration/team/list.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    cy.contains('Welcome to SalesWell')
+  })
+  it('Should have permissions when logged in as user', () => {
+    cy.login('normal@example.com', 'password')
+    cy.visit('http://localhost:3000/team-members/list')
+
+    cy.contains('admin user').should('exist')
```
Comment (keepcoding-tech) [inline cypress/integration/team/list.js]: You should be testing all 10 members, one by one

File: prisma/seedTeamMembers.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { PrismaClient, Role } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+// This function will create 10 team members for testing the list of prospects
```
Comment (keepcoding-tech) [inline prisma/seedTeamMembers.ts]: `the list of prospects` => `the list of team members`

File: prisma/seedTeamMembers.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    }
+    for (let index = 6; index <= 10; ++index) {
+      await prisma.team_members.create({ data: {
+        name: 'Team Member ' + index,
+        mail: 'admin_member' + index + '@gmail.com',
+        password: 'sdwa' + index,
+        role: Role.ADMIN,
+      } })
+    }
+  }
```
Comment (keepcoding-tech) [inline prisma/seedTeamMembers.ts:16]: Indentation :(

File: pages/team-members/list.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    getMorePageData(getMorePageDataParams)
+  }, [])
+
+  return (
+    <>
+      <InfiniteScroll
+        dataLength={teamMembers.length}
+        next={() => {getMorePageData(getMorePageDataParams)}}
+        hasMore={hasMore}
+        loader={<h3> Loading...</h3>}
```
Comment (keepcoding-tech) [inline pages/team-members/list.tsx]: you have an extra space here

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

**Comment by keepcoding-tech** [comment]
Let's be consistent with the naming and rename the file to `cypress/integration/team-members`

File: cypress/integration/team/list.js (hunk 2 above)
Comment (CiprianDumitriu) [reply cypress/integration/team/list.js]: Nu stiu exact la ce te referi dar voi presupune ca la Welcome to Saleswell

Comment (CiprianDumitriu) [reply cypress/integration/team/list.js]: I don't know why it's wrong. If we don't have access to /team-members/list we will go back to Welcome to Saleswell and we won't find any elements on that page that we searched for.

Comment (keepcoding-tech) [reply cypress/integration/team/list.js]: Yes, but the logic should be that you check the things you see, in which case you should also check that the "welcome" page has loaded.

File: cypress/integration/team-members/list.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  
+  it('Should have permissions when logged in as user', () => {
+    cy.login('normal@example.com', 'password')
+    cy.visit('http://localhost:3000/team-members/list')
+
+    cy.contains('admin user').should('exist')
+    cy.contains('admin@example.com').should('exist')
+    cy.contains('ADMIN').should('exist')
+
+    cy.contains('Team Member 10').should('exist')
```
Comment (keepcoding-tech) [inline cypress/integration/team-members/list.js]: Let's use a `for` loop here

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: cypress/integration/team-members/list.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  
+  it('Should have permissions when logged in as user', () => {
+    cy.login('normal@example.com', 'password')
+    cy.visit('http://localhost:3000/team-members/list')
+
+    cy.contains('admin user').should('exist')
+    cy.contains('admin@example.com').should('exist')
+    cy.contains('ADMIN').should('exist')
+
+    for (let i = 10; i >= 1; i--) {
```
Comment (keepcoding-tech) [inline cypress/integration/team-members/list.js]: Let's make two `for` loops, one for normal users and one for admin users, is more clear. Also, let's use some comments.

File: prisma/seedTeamMembers.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import { PrismaClient, Role } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+// This function will create 10 team members for testing the list of team members
```
Comment (keepcoding-tech) [inline prisma/seedTeamMembers.ts]: Make sure you have 80 characters max on each line, also let's use some comments above each `for` loop

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: cypress/integration/team-members/list.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  it('Should have permissions when logged in as user', () => {
+    cy.login('normal@example.com', 'password')
+    cy.visit('http://localhost:3000/team-members/list')
+
+    cy.contains('admin user').should('exist')
+    cy.contains('admin@example.com').should('exist')
+    cy.contains('ADMIN').should('exist')
+
+    // This is the test for team members with ADMIN role
+    for (let i = 10; i >= 6; i--) {
```
Comment (keepcoding-tech) [inline cypress/integration/team-members/list.js]: Why are you decreasing from 10 to 6 and not going up right from 6 to 10? Alos, let's check the role on  the card too.

File: cypress/integration/team-members/list.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+    cy.scrollTo('bottom')
+
+    // This is the test for team members with USER role
+    for (let i = 5; i >= 1; i--) {
+      cy.contains(`Team Member ${i}`).should('exist')
+      cy.contains(`team_member${i}@gmail.com`).should('exist')
+    }
+    
+    cy.contains('USER').should('exist')
```
Comment (keepcoding-tech) [inline cypress/integration/team-members/list.js:39]: Check the user's name and email too

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: pages/team-members/list.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
+import getMorePageData from '../../lib/get-more-page-data'
+import InfiniteScroll from 'react-infinite-scroll-component'
+import TeamMemberCard from '../../components/team-members/TeamMemberCard'
+import { useEffect, useState } from 'react'
+
+function TeamMembersList() {
```
Comment (S7012MY) [inline pages/team-members/list.tsx:6]: we should have the naming consistent with the file

**Review by S7012MY — CHANGES_REQUESTED** [review]

**Review by S7012MY — APPROVED** [review]

**Review by keepcoding-tech — APPROVED** [review]

**Pull request merged**